src/metabase/models/collection.clj

```diff
@@ -585,10 +587,16 @@
                                          [:= :p.perm_value "read-and-write"]
                                          (when (= :read (:permission-level visibility-config))
                                            [:= :p.perm_value "read"])]]}
-                               (when-let [personal-collection-and-descendant-ids (user->personal-collection-and-descendant-ids api/*current-user-id*)]
+                               {:select [[:c.*]]
+                                :from [[:collection :c]]
+                                :where [:= :type "trash"]}
```

Comment:
Should we add an index on `collection.type`? I'm a little worried this would be slow if an instance has a huge number of collections otherwise since it'll be a linear scan over every collection. (But I haven't tested it)

Reply:
That's a good point - I think that would make sense.